File: src/main/scala/beam/physsim/DummyPhysSim.scala
package beam.physsim

import akka.actor.Props
import beam.agentsim.sim.AgentsimServices
import beam.physsim.PhysSim.{InitializePhysSim, PhysSimInitialized}

/**
  * BEAM
  */
class DummyPhysSim (agentsimServices: AgentsimServices) extends PhysSim {

  override def receive: Receive = {
    case InitializePhysSim =>
      log.info("Initializing DummyPhysSim")
      sender() ! PhysSimInitialized
    case msg =>
      log.info(s"Unknown message received by DummyPhysSim: $msg")
  }
}

object DummyPhysSim {
  def props(agentsimServices: AgentsimServices) = Props(classOf[DummyPhysSim], agentsimServices)
}

File: src/main/java/beam/physsim/DummyPhysSim.java
package beam.physsim;

import akka.actor.Props;
import akka.actor.UntypedActor;
import beam.sim.BeamServices;

/**
 * BEAM
 */
public class DummyPhysSim extends UntypedActor{
    BeamServices services;

    public DummyPhysSim(BeamServices services) {
        this.services = services;
    }

    public static Props props(BeamServices services){
        return Props.create((Class<?>) DummyPhysSim.class,services);
    }
    @Override
    public void onReceive(Object message) throws Throwable {
        if(message instanceof InitializePhysSim){
            System.out.println("Init Phys Sim");
            getSender().tell(new PhysSimInitialized(),getSelf());
        }
    }
}
